### PR TITLE
add test for Dag::Read::get_chunk

### PR DESCRIPTION
**Pull request by phritz**
progress towards #35

### CODE REVIEW

File: src/dag/read.rs [hunk 1]
```diff
@@ -63,30 +63,76 @@ mod tests {
...
+        test("present", false, true).await;
+        test("present", true, true).await;
+        test("no such hash", false, false).await;
+    }
+
+    #[async_std::test]
+    async fn test_get_chunk() {
+        async fn test(data: Vec<u8>, refs: &[&str], get_same_chunk: bool) {
+            let kv = MemStore::new();
+            let cloned_data = data.clone();
```
Comment (aboodman) [inline src/dag/read.rs:98]: [cbb] Eliminate this local by calling `clone` inline at 100.

File: src/dag/read.rs [hunk 2]
```diff
@@ -63,30 +63,76 @@ mod tests {
...
+                        refs,
+                        got_chunk.refs().unwrap().collect::<Vec<&str>>().as_slice()
+                    );
+                }
+            } else {
+                let got_option = r.get_chunk("no such hash").await.unwrap();
+                assert_eq!(None, got_option);
+            }
+        }
+        test(vec![1u8], &vec!["r1", "r2"], true).await;
```
Comment (aboodman) [inline src/dag/read.rs:134]: You can say `vec![1]` (the compiler knows it must be u8 due to parameter type of `test`).

File: src/dag/read.rs [hunk 3]
```diff
@@ -63,30 +63,76 @@ mod tests {
...
+                .await
+                .unwrap();
+            if let Some(meta) = chunk.meta() {
+                kvw.put(&Key::ChunkMeta(chunk.hash()).to_string(), meta)
+                    .await
+                    .unwrap();
+            }
+            kvw.commit().await.unwrap();
+
+            // TODO(phritz) start a write tx here that completes before r.get_chunk
```
Comment (aboodman) [inline src/dag/read.rs:111]: 🙏

File: src/dag/read.rs [hunk 4]
```diff
@@ -63,30 +63,76 @@ mod tests {
...
+            // below and ensure it does not affect the result.
+            let kvr = kv.read().await.unwrap();
+            let r = Read { kvr };
+            // TODO(phritz) start a write tx here that completes before r.get_chunk
+            // and ensure it does not affect the result.
+            if get_same_chunk {
```
Comment (aboodman) [inline src/dag/read.rs:117]: [minifix]

Chunk implements `PartialEq` so I believe that we can reduce this entire if/else block to something like:

```
let expected: Option<Chunk>::None;
if get_same_chunk {
  expected = Some(chunk);
}

assert_eq!(chunk, r.get_chunk(chunk_hash).await.unwrap());
```

File: src/dag/read.rs (hunk 3 above)
Comment (phritz) [reply src/dag/read.rs:111]: well, actually... see incoming PR

File: src/dag/read.rs (hunk 4 above)
Comment (phritz) [reply src/dag/read.rs:117]: i tried for 15m to take your suggestion but i can't figure out how to make the compiler happy about getting chunk.hash() out of the chunk so we can use it in get_chunk. if you can make this work lemme know i would like to see solution. (also i can post my attempt if you want.)

Comment (aboodman) [reply src/dag/read.rs:117]: A way to write it that is close to what I suggested it:

```
            let mut expected = Option::<Chunk>::None;
            let chunk_hash: &str;
            if get_same_chunk {
                expected = Some(chunk);
                chunk_hash = expected.as_ref().unwrap().hash();
            } else {
                chunk_hash = "no such hash";
            }
            assert_eq!(expected, r.get_chunk(chunk_hash).await.unwrap());
```

Here is a perhaps simpler way:

```
            if get_same_chunk {
                let expected = Some(chunk);
                assert_eq!(
                    expected,
                    r.get_chunk(expected.as_ref().unwrap().hash())
                        .await
                        .unwrap()
                );
            } else {
                assert_eq!(None, r.get_chunk("No such chunk").await.unwrap());
            }

```

Comment (aboodman) [reply src/dag/read.rs:117]: I'm sure there's smaller versions (maybe by making `expected` contain a ref), but running out the door :).

Comment (phritz) [reply src/dag/read.rs:117]: thanks, went with your first one which is easiest for me to read. i was missing the as_ref() to make it work!

**Pull request merged**